sycl/plugins/level_zero/CMakeLists.txt

```
@@ -99,9 +99,13 @@ add_sycl_plugin(level_zero
     Threads::Threads
     ${XPTI_LIBS}
 )
-
 add_dependencies(pi_level_zero ze-api)
 
+if (WIN32)
+ target_link_libraries(pi_level_zero PRIVATE delayimp)
+ target_link_options(pi_level_zero PRIVATE /DELAYLOAD:ze_loader.dll)
```

can we somehow use `LevelZeroLoader` as above at https://github.com/intel/llvm/pull/11902/files#diff-e162c7afd5dbcb57ced84354c427518f0f37b190538d136e871855d78975533bR98 and not hardcode `ze_loader.dll`?

---

Passing LevelZeroLoader to /DELAYLOAD doesn't seem to work, it needs the exact name of the DLL.

---

[target_link_options](https://cmake.org/cmake/help/latest/command/target_link_options.html) supports generator expressions and `LevelZeroLoader` is a target, try [TARGET_FILE](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_FILE) like this:

```suggestion
 target_link_options(pi_level_zero PRIVATE /DELAYLOAD:$<TARGET_FILE:LevelZeroLoader>)
```

---

Thanks for the suggestion @kbenzie! However, i get a build error:  Error evaluating generator expression:
 $<TARGET_FILE:LevelZeroLoader>
Target "LevelZeroLoader" is not an executable or library.